lib/msf/core/module/external.rb

```diff
@@ -5,11 +5,6 @@ module Msf::Module::External
 
   def execute_module(path, method: :run, args: datastore, fail_on_exit: true)
     mod = Msf::Modules::External.new(path, framework: framework)
-    if args.is_a?(Msf::DataStore) || args.is_a?(Msf::DataStoreWithFallbacks)
```

Comment:
I originally added this in https://github.com/rapid7/metasploit-framework/pull/17490 - but I believe this PR is a cleaner solution

The real issue is the external python modules use a datastore that doesn't have the same semantics as the Ruby version; i.e. i.e. RHOST/RHOSTS aliasing support, plus case insensitive lookups